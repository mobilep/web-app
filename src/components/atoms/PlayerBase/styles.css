.PlayerBase {
	background-color: var(--black);
	height: auto;
	max-height: 100%;
	width: 100%;
}

.PlayerBase video {
	width: 100%;
	height: 100%;
}
