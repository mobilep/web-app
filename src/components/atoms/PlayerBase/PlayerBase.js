import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import shaka from 'shaka-player';
import './styles.css';

export default class PlayerBase extends Component {

	static propTypes = {
		autoPlay: PropTypes.bool,
		className: PropTypes.string,
		fallBackStreamUrl: PropTypes.string.isRequired,
		onStreamError: PropTypes.func,
		onStreamLoad: PropTypes.func,
		streamUrl: PropTypes.string.isRequired,
	}

	static defaultProps = {
		autoPlay: true,
		onStreamLoad: () => {},
		onStreamError: () => {},
	}

	el = React.createRef();

	componentDidMount () {
		this.setupPlayer();
	}

	isSupportedByPlayer (src) {
		return src.includes('m3u8') || src.includes('mpd');
	}

	setupPlayer () {
		const { streamUrl, fallBackStreamUrl } = this.props;
		const videoEl = this.el.current || {};

		if (!this.isSupportedByPlayer(streamUrl)) {
			videoEl.src = streamUrl;
			return;
		}

		if (window.MediaSource) {
			shaka.polyfill.installAll();
			this.player = new shaka.Player(videoEl);
			this.player
				.load(streamUrl)
				.then(this.onStreamLoad)
				.catch(this.onStreamError);
		} else {
			videoEl.src = fallBackStreamUrl;
		}
	}

	onStreamLoad = () => {
		this.props.onStreamLoad();
	}

	onStreamError = (error) => {
		// eslint-disable-next-line
		console.error('Error code', error.code, 'object', error);
		this.props.onStreamError();
	}

	getClassName ({ className } = this.props) {
		return classNames('PlayerBase', className);
	}

	render () {
		const { autoPlay } = this.props;
		return <video autoPlay={autoPlay} className={this.getClassName()} controls ref={this.el} />;
	}

}
