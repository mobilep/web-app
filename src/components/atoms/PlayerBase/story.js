import React from 'react';
import { storiesOf } from '@storybook/react';
import PlayerBase from './PlayerBase';

storiesOf('1. Atoms / PlayerBase', module)
	.add('PlayerBase', () => {
		const streamUrl = 'https://s3-eu-west-1.amazonaws.com/mobilepractice-video/MPEG-DASH/79E5390D-0DA0-4929-B753-38BAE1271173/playlist.mpd';
		return <PlayerBase streamUrl={streamUrl} />;
	});
