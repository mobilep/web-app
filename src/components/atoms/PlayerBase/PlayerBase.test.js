import React from 'react';
import { shallow } from 'enzyme';

import PlayerBase from './PlayerBase';

describe('Atom - PlayerBase', () => {

	it('Should render', () => {
		const wrapper = shallow(<PlayerBase streamUrl='url' fallBackStreamUrl='url2' />);
		// An error is logged as Shaka Player can't start (because we're rendering in a test environment).
		// eslint-disable-next-line no-console
		console.error = () => {};
		expect(wrapper).toMatchSnapshot();
	});

});
