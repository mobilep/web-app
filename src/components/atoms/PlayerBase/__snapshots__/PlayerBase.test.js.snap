// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - PlayerBase Should render 1`] = `
<video
  autoPlay={true}
  className="PlayerBase"
  controls={true}
/>
`;
